components/raftstore/src/store/fsm/peer.rs

```diff
@@ -4545,6 +4551,31 @@ where
         self.register_pd_heartbeat_tick();
     }
 
+    fn on_refresh_region_buckets(&mut self, region_epoch: RegionEpoch, bucket_keys: Vec<Vec<u8>>) {
+        let region = self.fsm.peer.region();
+        if util::is_epoch_stale(&region_epoch, region.get_region_epoch()) {
+            info!(
+                "receive a stale refresh region bucket message";
+                "region_id" => self.fsm.region_id(),
+                "peer_id" => self.fsm.peer_id(),
+                "epoch" => ?region_epoch,
+                "current_epoch" => ?region.get_region_epoch(),
+            );
+            return;
+        }
+        let mut region_buckets = Buckets::default();
+        let now = monotonic_raw_now();
+        const NANOSECONDS_PER_SECOND: u64 = 1_000_000_000;
+        region_buckets.version = (now.sec as u64) * NANOSECONDS_PER_SECOND + now.nsec as u64;
+        region_buckets.set_keys(bucket_keys.into());
+        // add region's start/end key
+        region_buckets
```

Comment:
Maybe using `VecDeque` for `buckets_keys` could reduce the cost of constructing `keys`.